Hook in promtail converter CLI & fix issues (#4674)
grafana/agent: docs/sources/flow/reference/cli/convert.md

* `--report`, `-r`: The filepath and filename where the report is written.

> * `--source-format`, `-f`: Required. The format of the source file. Supported formats: [prometheus], [promtail].

Reviewer: I think the intent of the `[]` syntax was to be:

```suggestion
* `--source-format`, `-f`: Required. The format of the source file. Supported formats: [prometheus, promtail].
```

Author: it was meant to be a link to an anchor, but indeed something was wrong - I forgot to add the link reference.

Reviewer: Ah, my bad, thanks!

and many supported *_sd_configs. Unsupported features in a source config result
in [errors].

### Promtail

Using the `--source-format=promtail` will convert the source config from
[Promtail v2.8.x](https://grafana.com/docs/loki/v2.8.x/clients/promtail/)
to Grafana Agent Flow config.

> Nearly all [Promtail features](https://grafana.com/docs/loki/v2.8.x/clients/promtail/configuration/)

Author: It may not be practical to list all of them. One reason is that we don't have a list of all features that promtail has to start with and some are supported but not automatically converted - require manual River config changes. Another class of features that we don't support in promtail are undocumented. Furthermore, when we add support for new features, having a list here may quickly get outdated. 

So I would prefer not to list the supported/unsupported features here, but encourage people to run the conversion and see in a report whether something is not supported for their case. 

Do you have a suggestion how we could capture this in the doc? Or maybe you think that having a list of features is really necessary even if risking it going out of date?

> Nearly all [Promtail features](https://grafana.com/docs/loki/v2.8.x/clients/promtail/configuration/)

Reviewer 2: Yeah capturing all the unsupported features could really be a challenge in this case. This was just a question that popped into my head as I read the text here... 

Leave it as is... it's just an observation.

are supported and can be converted to Grafana Agent Flow config.

> Unsupported features in a source config result in [errors]. The converter will

Reviewer 2: ```suggestion
If you have unsupported features in a source configuration, you will receive [errors] when you convert to a flow configuration.. The converter will
```

> If you have unsupported features in a source configuration, you will receive [errors] when you convert to a flow configuration.. The converter will

Reviewer 2: ```suggestion
If you have unsupported features in a source configuration, you will receive [errors] when you convert to a flow configuration. The converter will
```

Remove extra full stop that showed up during edits
